packages/devtools_flutter/lib/src/connect_screen.dart

```diff
@@ -98,9 +103,26 @@ class _ConnectScreenBodyState extends State<ConnectScreenBody> {
     );
   }
 
-  void connect() {
-    // TODO(https://github.com/flutter/devtools/issues/1112): Connect to a VM.
-    Navigator.pushNamed(context, '/connected');
+  Future<void> connect([_]) async {
+    final connected = await service.FrameworkCore.initVmService(
+          '',
+          explicitUri: Uri.parse(controller.text),
+        ).catchError((_) {
```

Comment:
why are we using catchError instead of try catch here?

Reply:
Caught.